### ISSUE TITLE
Ubuntu 24.04 image available to the public (and how to make images)

### ISSUE THREAD

**Issue by knikolla** [CCI-MOC/esi#608]
The current MOC ESI environment has a ubuntu 22.04 image available, but not a 24.04 one.

The [disk image builder documentation](https://docs.openstack.org/ironic/latest/user/creating-images.html) discourages using partition images, which is what the [guide for creating the 22.04 image used](https://github.com/CCI-MOC/esi/issues/589#issuecomment-2276921958).

Based on the guide it also seems possible that the [cloud image distributed by Ubuntu](https://cloud-images.ubuntu.com/noble/current/) might just work (since it should support EFI booting). 

[Link to slack thread](https://massopencloud.slack.com/archives/CG1C4SZCM/p1724861192204599)

**Comment by skanthed** [comment]
@tzumainn @knikolla 

I chose partitioning for Ubuntu 22.04 because the existing Ubuntu images and other images in our environment were also partition images. Partitioning provides several advantages, such as greater flexibility in deployment, especially in environments where network booting (like PXE) is used, I thought we were using systems that might require different configurations. This approach aligns with the setup that was already there. 

I have created the whole disk image too for centos9-stream - https://github.com/CCI-MOC/esi/issues/564
If you want I can also create whole disk image for 24.04 or lay down the steps for doing it.

**Comment by knikolla** [comment]
@skanthed I think it's worth documenting the advantages and disadvantages of both approaches as they apply to our environment and the steps for doing either. For example, what is different about network booting with whole disk vs partition images? When would a user want to create or use one image versus the other? This information is scarce in upstream Ironic too. 

Also, testing if the stock whole disk Ubuntu image distributed by Ubuntu works on our environment.

**Comment by knikolla** [comment]
(By documentation I mean updating esi.readthedocs.io)

**Comment by joachimweyl** [comment]
@skanthed please update this issue with the current status.

**Comment by skanthed** [comment]
I began testing Ubuntu 24.04 yesterday with both cloud-init and without it to evaluate the functionalities and behavior of the image. I will provide feedback as soon as the tests are completed.

**Comment by skanthed** [comment]
@tzumainn Image is ready for review - `ubuntu-24.04` in my private repo. I am working on document will attach it here.

**Comment by tzumainn** [comment]
The image isn't quite working for me - metalsmith will "complete", but there are errors in the console, and the node won't boot from the hard drive afterwards. If you're using diskimage-builder, are you including the `block-device-efi` element?

**Comment by skanthed** [comment]
No, I am not using any` block-devie-efi` element.  Yes metalsmith completed and I was able to ssh and everything looked fine. I have one alternate image with little bit different setup. I will upload it and you can try that and let me know if that works.

**Comment by skanthed** [comment]
@tzumainn Can you test this image - `test-ubuntu-24.04` and let me know if this works?

**Comment by morantara** [comment]
@skanthed was the above test completed?

**Comment by skanthed** [comment]
For now this issue has been moved to backlog due to some high priority issues.
